# 인사

안녕하세요 ~~ 25기 여러분 반갑습니다 ~~  

드디어 정규세션이 시작되었네요.  

뭘 과제로 내볼까 고민하다가, 요즘 데이터 분석 채용 공고 자격 요건에 필수로 들어있는.... A/B TEST가 좋겠다!!고 생각했습니다.  

데이터 분석쪽이 아니더라도, 다양한 도메인에서 쓸 수 있는 방법이라고 생각해요.


A/B TEST의 개념부터 기초적인 과정을 이해해볼 수 있게 내봤어요. 근데 설명해야할 게 많다보니 텍스트가 너무 많아졌네요.

읽기 힘들어도 ㅎㅎ... 열심히 해주시고, 질문 사항은 저에게 연락 남겨주세요.  

참고로, 정규세션 발표 외의 내용을 구글링하고 공부해야만 풀 수 있는 문제들이 포함되어 있습니다.

제 이름... 아시죠...? 24기 박현서입니다....



<br>

# Q1. A/B TEST란 무엇일까요?   
또한, A/B TEST시 유의할 사항들을 초두효과를 포함해 기술해주세요.

정규세션에서 발표한 내용 외에 추가적으로 공부한 개념을 작성하셔도 됩니다.
정의와 함께 예시 하나만 들어주세요 ~~

## A1. 답안 작성  
답변:
A/B TEST란 웹사이트, 앱 등에서 두 가지 버전의 콘텐츠를 무작위 사용
자 집단에 노출시켜 어떤 버전이 더 나은 성과를 내는지 비교하여 둘 중 더 나은 버전을 통계적인 기반으로 선택하기 위한 실험이다.

A/B TEST 시 초두효과에 유의해야한다. 이는 실제 웹사이트, 앱 등의 환경이 바뀌었을 때 사용자가 기존의 버전(A)에 비해 새로운 변화(B)가 낯설게 느껴져 클릭율 등이 저조하게 나오는 것을 의미한다.

초두효과와 반대되는 효과로 신기함 효과가 있다. 사용자가 오히려 새로운 변화에 흥미를 느끼고 클릭해 발생하는 효과이다.

예를 들어, 한 무신사 어플에서 상품 가격을 빨간색으로 바꾸는 A/B TEST를 진행한다고 치자. 사용자 집단 m에서는 빨간 가격표가 낯설어 앱을 떠나는 비율이 높은 초두효과가 발생하고, 사용자 집단 n에서는 신기함 효과가 발생해 더 많은 클릭이 발생했다. 이런 효과들을 최소화하기 위해 실험 기간을 길게 두어 실험을 진행하여야 한다.



---



 A/B TEST를 하기 위해서는 1-pager를 작성해야합니다.  

1-pager란 문제 정의, 관찰하고자하는 실험대상, 가설설정, 평가지표 등을 작성하는  
간단히 말해 계획서를 의미하는데요,

다른 부서와의 소통을 위한 역할 또,과거에 어떤 테스트를
어떻게 진행했는지에 대한 기록의 역할을 하기도 합니다.


 A/B TEST의 1-pager는 복잡한 실험 과정을 한눈에 파악하고, 의사결정권자가 "그래서 이 실험을 왜 했고, 결과가 무엇이며, 무엇을 할 것인가?"를 바로 이해할 수 있도록 작성하는 것이 핵심입니다.

예시를 통해 어떤 항목이 있고, 각 항목에 어떤 내용을 채우는 건지 보여드릴게요.



---



**예시**   
(푸는 거 아니고, 예시입니다! 읽으면서 이런 부분은 이런 내용을 채우는 거군... 이해하시면 돼요!)  




 📊 [ThinkGood] 메인 홈 검색 모듈 상위 노출 실험 1-Pager

---

**1. 실험 전략 및 배경 (Strategy & Observation)**

| 항목 | 내용 |
| :--- | :--- |
| **Target User** | 특정 공모전을 찾으려는 목적이 뚜렷하지만, 현재 하단에 위치한 검색바를 즉각적으로 발견하지 못한 유저 |
| **Observation** | **[데이터 분석 결과]** <br> • 검색 이용 유저의 상세 진입 전환율(CVR)은 40%, 미이용 유저의 전환율은 25%임 <br> • 현재 검색 이용률은 30%로, 검색 가시성 개선을 통한 전환율 상승 여력이 큼 |
| **Problem Statement** | 검색바가 배너 하단에 위치하여 시인성이 낮으며, 유저가 정보를 탐색하기 전 이탈할 확률이 높음 |
| **Hypothesis** | 하단 검색바를 페이지 **최상단으로 노출**시켜 검색 접근성을 높인다면, 전체 공모전 상세 진입률(CVR)이 개선될 것이다 |

---

 **2. 실험군 설계 (Experimental Groups)**

<table>
  <tr>
    <td align="center" width="50%">
      <b>A (Control) - 기존안</b><br>
      <img src="https://i.postimg.cc/kgcLc7Qy/Gemini-Generated-Image-enqvuhenqvuhenqv-(1).png" width="100%"><br>
      <sub>검색바 하단 위치 (기존 UI)</sub>
    </td>
    <td align="center" width="50%">
      <b>B (Variant) - 실험안</b><br>
      <img src="https://i.postimg.cc/0N51Rz5M/Gemini-Generated-Image-enqvuhenqvuhenqv.png" width="100%"><br>
      <sub>검색바 최상단 배치 (변경 UI)</sub>
    </td>
  </tr>
</table>

---

 **3. 지표 및 통계 설계 (Metrics & Statistics)**

> **핵심 측정 지표**
> * **Primary Metric (주지표):** 공모전 상세 페이지 진입률 (Detail Page CVR)
> * **Secondary Metric (보조지표):** 검색바 클릭률 (Search CTR)
> * **Guardrail Metric (방어지표):** 메인 페이지 이탈률 (Bounce Rate)

| 통계적 설계 항목 | 설정값 | 비고 |
| :--- | :--- | :--- |
| **Significance Level ($\alpha$)** | 5% | 유의수준 0.05 |
| **Statistical Power ($1-\beta$)** | 80% | 검정력 유지 |
| **Minimum Sample Size** | **총 92,000명** | 각 그룹당 46,000명 할당 |
| **Required Period** | **최소 4일** |  

---

**4. 위험 관리 및 중단 기준 (Risk & Andon)**

* **Trade-Off:**   
    검색바 상단 노출로 인해 메인 배너의 시각적 주목도가 분산될 수 있으며, 기존 UI에 익숙한 유저에게 일시적인 초두 효과(Primacy Effect)에 의한 혼란이 발생할 수 있음.
* **Andon (중단 기준):**
    1. 대조군 대비 전체 상세 진입률이 **5% 이상 유의미하게 하락**할 경우
    2. UI 변경으로 인한 페이지 오류나 유저 이탈률이 **3일 연속 급증**할 경우

---







<br>

# Q2. 1-pager를 작성해봅시다. (꼬리문제 3 + 보고서 채우는 것1)
저희가 만든 데모 사이트 링크를 첨부해드릴테니, 마음껏 보신 뒤에 위의 예시처럼 1-pager를 작성해보기 바랍니다.

여기서 실험군 설계에 들어가는 사진은 굳이 넣을 필욘 없고 말로만 설명해주셔도 됩니다. 아니면 gemini에게 요청해볼 수도 있겠네요!  

또한 문제탐색에 필요한 cvr과 같은 지표들은 직접 가정하고 하시면 됩니다. 다만, 해당 지표들이 큰 문제가 있는 지표여야... a/b test를 진행하는 이유가 되겠죠!?


정리해보면, 저희 웹사이트 보시고 '이건 여기에 배치하면 더 좋을 거 같은데?' 하는 것들을 찾아서 모든 것을 가정한 뒤에 쓰시면 됩니다!

[데모 웹사이트](https://huggingface.co/spaces/HanJaeHui/ThinkGood_Recommendation_System)


추가적으로 아래 실험군 항목에 사진을 넣으실 분들은 제가 아래 첨부드리는 사이트에 방문하셔서 해당 이미지를 코랩에 넣을 수 있는 형태로 변환한 링크를 복붙하시면 됩니다.  

(.png로 끝나는 링크 복붙하시면 됩니다.)

[포스트 이미지 사이트](https://postimages.org/)



<br>

## Q2-1. 실험군을 설계 해봅시다.
아래 보고서를 채워 주세요.

## A2-1. 아래 보고서 항목 설명을 참고하여 '[ ]' 부분을 지우고 채워주세요.  
  



📊 [프로젝트명] 실험 기획안 1-Pager

---

1. 실험 전략 및 배경 (Strategy & Observation)

| 항목 | 내용 |
| :--- | :--- |
| **Target User** | • 스크롤을 끝까지 내리고 답답함을 느끼는 유저, <br> • 과거 공모전 데이터를 보고 싶은 유저 |
| **Observation** | **데이터 근거** <br> • 스크롤을 끝까지 내린 유저가 전체 유저의 10% <br> • 스크롤을 끝까지 내린 유저의  페이지 이탈율이 80% |
| **Problem Statement** | 페이지를 끝까지 내린 유저라면 공모전에 큰 관심이 있는 유저일텐데도 페이지 이탈율이 높다. <br>이는 더이상 내려가지 않는 화면에 답답함을 느끼거나 더 볼만한 공모전이 없다고 생각했을 가능성이 높다. |
| **Hypothesis** | **무한 스크롤 형태로 변환해 과거 공모전을 이어서 보여준다면 페이지 이탈율이 낮아질 것이다**  |

---

 2. 실험군 설계 (Experimental Groups)

<table>
  <tr>
    <td align="center" width="50%">
      <b>A (Control) - 기존안</b><br>
      <img src="[기존 시안 이미지 URL 입력]" width="100%"><br>
      <sub>페이지를 끝까지 내리면 더이상 스크롤인 내려가지 않는다.</sub>
    </td>
    <td align="center" width="50%">
      <b>B (Variant) - 실험안</b><br>
      <img src="[변경 시안 이미지 URL 입력]" width="100%"><br>
      <sub>현재 진행 중인 공모전을 모두 보여주고 난 후엔 과거 완료된 공모전들이 계속 이어진다. </sub>
    </td>
  </tr>
</table>


<br>

## Q2-2. 실험에 필요한 최소 test users 수를 구해봅시다.
유의수준 0.05, 검정력 80% 하에 수행해주세요.  


아래 공식 참고해서, 최소 test users 수를 구하는 코드 작성해주세요.

추가적으로 n은 산출하는데에 가정한 p1, p2도 같이 작성해주세요.

$$n = \frac{(Z_{\alpha/2} + Z_{\beta})^2 \cdot [p_1(1-p_1) + p_2(1-p_2)]}{(p_1 - p_2)^2}$$

## A2-2. 아래 질문에 대한 답과 코드를 채워주세요.

p1, p2는 각각 무엇을 의미하는 확률이고, 어떤 값으로 가정하셨나요?   
또한 위 공식을 보고 n을 구하기 위한 코드를 작성해주세요. 단, 라이브러리는 z값을 구하는 용도의 것만 사용해주세요.


### 답변:

* $p_1, p_2$의 의미와 가정된 값$p_1$ : 대조군(기존 안)의 전환율이다. 현재 디자인에서의 평균적인 성과 지표를 의미한다.
* $p_2$ : 실험군(Experimental Group, 변경 안)의 목표 전환율이다. 우리가 실험을 통해 기대하는 개선된 성과 지표이다.

**[가정된 값]** 예시 코드를 작성하기 위해 임의로 다음과 같이 설정했다.
* $p_1 = 0.20$ (20%): 기존 공모전 상세페이지의 클릭율이 20%라고 가정.
* $p_2 = 0.22$ (22%): 디자인 변경을 통해 클릭율이 2%p 상승할 것이라고 기대하여 22%로 가정.

In [ ]:
# A2-2 코드작성. 나온 n은 아래 Minimum Sample Size에 기재.

import math
from scipy import stats

def calculate_min_sample_size(alpha, power, p1, p2):
    """
    사용자가 제시한 공식을 사용하여 최소 표본 크기(n)를 계산하는 함수
    n = ((Z_alpha/2 + Z_beta)^2 * (p1(1-p1) + p2(1-p2))) / (p1 - p2)^2
    """

    # 1. Z값 구하기 (표준정규분포 역함수 사용)
    # 유의수준 alpha에 대한 임계값 (양측 검정, alpha/2)
    # alpha=0.05일 때 약 1.96
    z_alpha = stats.norm.ppf(1 - alpha / 2)

    # 검정력(1-beta)에 대한 임계값
    # Power=0.8일 때 약 0.84
    z_beta = stats.norm.ppf(power)

    # 2. 공식 대입
    # 분자: (Z_alpha + Z_beta)^2 * 분산항 합
    numerator = (z_alpha + z_beta)**2 * (p1 * (1 - p1) + p2 * (1 - p2))

    # 분모: (두 확률 차이)^2
    denominator = (p1 - p2)**2

    # 3. 최종 n 계산
    n = numerator / denominator

    return n, z_alpha, z_beta

# --- 설정값 입력 ---
alpha = 0.05  # 유의수준 5%
power = 0.80  # 검정력 80%
p1 = 0.20     # 대조군 전환율 가정 (10%)
p2 = 0.22     # 실험군 전환율 가정 (12%)

# 계산 실행
required_n, z_a, z_b = calculate_min_sample_size(alpha, power, p1, p2)

print(f"=== 입력 변수 ===")
print(f"유의수준(alpha): {alpha}")
print(f"검정력(power): {power}")
print(f"기존 전환율(p1): {p1}")
print(f"목표 전환율(p2): {p2}")
print(f"\n=== 중간 계산 값 ===")
print(f"Z_alpha/2: {z_a:.4f}")
print(f"Z_beta:    {z_b:.4f}")
print(f"\n=== 결과 ===")
print(f"그룹 당 최소 필요 표본 수 (n): {required_n:.4f}")
print(f"올림 처리 후 (n): {math.ceil(required_n)} 명")










#########

=== 입력 변수 ===
유의수준(alpha): 0.05
검정력(power): 0.8
기존 전환율(p1): 0.2
목표 전환율(p2): 0.22

=== 중간 계산 값 ===
Z_alpha/2: 1.9600
Z_beta:    0.8416

=== 결과 ===
그룹 당 최소 필요 표본 수 (n): 6506.7213
올림 처리 후 (n): 6507 명


<br>

## Q2-3. 실험에 필요한 최소 실험 기간을 구해봅시다.

이 실험에 필요한 최소 test users가 92000명이 나왔고  
해당 웹페이지 하루 평균 방문자수가 50000명 정도라고 할 때 우리는 최소 4일이 필요합니다.

비교하고 싶은 대상이 총 두개이고, 하나의 대상을 실험하기 위해선 최소 이틀은 필요하기 때문이죠!

위에서 구하신 test users수를 바탕으로, 하루 평균 방문자수를 가정한 뒤 기간을 설정해주세요.  
구하신 기간은 밑에 표에 기재해주시면 됩니다.



## A2-3. 아래 세질문에 답하신 뒤 아래 보고서를 채워주세요.

가정한 하루 평균 방문자 수?  2000명

산출된 최소 test users 수? 6507명
  
그래서 필요한 최소 기간은? 7일

## **아래 보고서도 채워주세요!**  
**'[ ]' 부분을 지우고 해당 내용에 부합하는 내용들을 채워주시면 됩니다.**

---

**3. 지표 및 통계 설계 (Metrics & Statistics)**

> **핵심 측정 지표**
> * **Primary Metric (주지표):**  공모전 카드 클릭률 (CTR)
> * **Secondary Metric (보조지표):** 공모 평균 페이지 체류 시간 (Average Session Duration)
> * **Guardrail Metric (방어지표):** 페이지 로딩 속도 (Page Load Time)

| 통계적 설계 항목 | 설정값 | 비고 |
| :--- | :--- | :--- |
| **Significance Level ($\alpha$)** | 5% | 보통 5%(0.05) 설정 |
| **Statistical Power ($1-\beta$)** | 80% | 보통 80% 설정 |
| **Minimum Sample Size** | **그룹당 6500명, 총 13000명** | 그룹당 인원수 명시 |
| **Required Period** | **최소 7일** | 실험 진행 예정 기간 |

---

 **4. 위험 관리 및 중단 기준 (Risk & Andon)**

* **Trade-Off :**    
    무한 스크롤로 인해 페이지 로딩 속도가 느려질 수 있음

* **Andon (중단 기준) :**
    1. 주지표 3% 이상 하락시 실험 중단
    2. UI 변경으로 인한 페이지 오류나 유저 이탈률이 **2일 연속 급증**할 경우 실험 중단

---

<br>

# Q3. A/B TEST를 수행해봅시다.

1-pager를 작성했으면 직접 test도 수행해봐야겠죠!  
위에 제가 만들었던 1-pager를 바탕으로 검정을 해볼게요.  
대표적인 검정방법에는 t-test가 있고 카이제곱 겁정이 있습니다.  


---



[실험 가설]  
귀무가설 ($H_0$): 메인 홈 검색 모듈의 위치(하단 vs 최상단)와 상세 페이지 진입 여부 사이에는 상관관계가 없다. 즉, 두 그룹의 상세 페이지 진입률($CVR$)은 동일하다.$$H_0: p_A = p_B$$대립가설 ($H_1$): 메인 홈 검색 모듈을 최상단에 노출하는 것은 상세 페이지 진입 여부에 영향을 미친다. 즉, 실험군(B)의 상세 페이지 진입률이 대조군(A)보다 높을 것이다.$$H_1: p_A < p_B$$


<br>

## Q3-1. 위 가설에 적합한 검정 이름과 왜 해당 검정을 써야하는지 이유도 작성해주세요. (t test와 카이제곱검정 중)

## A3-1.  검정 이름과 이유를 설명해주세요.

답변: 위 가설에 필요한 검정은 카이제곱 검정  이고,

이유: 분석하려는 결과 데이터인 '클릭 여부(Click / No Click)'가 연속된 숫자가 아닌 **범주형 데이터(Categorical Data)**이기 때문입니다.

우리는 두 그룹의 '평균 점수(Mean)'를 비교하는 것이 아니라, 클릭한 사람의 '빈도(Frequency) 혹은 비율'이 서로 다른지를 확인해야 하므로, t-test가 아닌 카이제곱 검정을 사용해야 합니다.

<br>

In [2]:
'''''''''''''''''''''''''''''''''''''''
제가 세운 A/B TEST 설계 실험 코드입니다.
'''''''''''''''''''''''''''''''''''''''

import pandas as pd
from scipy.stats import chi2_contingency

# 1. 관측 데이터 입력 (실제 실험 결과 수치로 변경하세요)
# [상세진입 성공 수, 상세진입 실패 수]
control_results = [11500, 34500]  # 대조군(A)
variant_results = [12190, 33810]  # 실험군(B)

# 2. 분할표(Contingency Table) 생성
data = [control_results, variant_results]

# 3. 카이제곱 검정 수행
chi2, p_value, dof, expected = chi2_contingency(data)

# 4. 결과 출력
print(f"카이제곱 통계량: {chi2:.4f}")
print(f"P-value: {p_value:.10f}")

alpha = 0.05
if p_value < alpha:
    print(f"결과: P-value가 {alpha}보다 작으므로 귀무가설을 기각합니다.")
    print("=> 검색바 최상단 노출이 상세 페이지 진입률에 유의미한 변화를 주었습니다.")
else:
    print(f"결과: P-value가 {alpha}보다 크므로 귀무가설을 기각할 수 없습니다.")
    print("=> 검색바 위치 변경의 효과가 통계적으로 증명되지 않았습니다.")

카이제곱 통계량: 26.9884
P-value: 0.0000002047
결과: P-value가 0.05보다 작으므로 귀무가설을 기각합니다.
=> 검색바 최상단 노출이 상세 페이지 진입률에 유의미한 변화를 주었습니다.


실제 데이터가 없어서, 저도 필요한 데이터는 가정했답니다.  

이제 25기 분들의 A/B TEST를 수행해보시죠!

<br>

## Q3-2. Q2에서 작성한 실험 수행을 위해 가설을 설정해주세요.


## A3-2. 아래 두 가설 내용을 채워주세요.
  
[실험 가설]  
귀무가설 ($H_0$): 무한 스크롤 기능 도입 여부와 공모전 카드 클릭 여부 사이에는 상관관계가 없다.<br>
즉, 기존(A)과 무한 스크롤(B) 그룹의 공모전 카드 클릭률($p$)은 동일하다. ($p_A = p_B$)
  <br><br>
대립가설 ($H_1$): 무한 스크롤 기능을 도입하는 것은 공모전 카드 클릭 여부에 영향을 미친다.<br>
즉, 무한 스크롤(B) 그룹의 공모전 카드 클릭률이 기존(A)보다 높을 것이다. ($p_A < p_B$)

<br>

## Q3-3. 위 가설을 수행하기 위해 어떤 검정 방법을 사용해야하고 이유는 무엇인지 작성해주세요.

## A3-3. 아래 검정 이름과 이유를 채워주세요.
답변: 위 가설에 필요한 검정은 카이제곱 검정이고,

우리가 확인하려는 데이터는 연속된 수치가 아니라, 사용자가 더 노출된 공모전 카드를 '클릭했는가/ 클릭하지 않았는가'로 나뉘는 범주형 데이터이기 때문이다. 두 그룹 간의 비율 차이를 검증하기 위해서는 t-test가 아닌 카이제곱 검정이 적합하다.

<br>

## Q3-4. 위에서 답한 것을 토대로, 검정을 수행해주세요. (t test와 카이제곱 검정 중)


## A3-4 아래 셀에 코드 작성해주세요. 그리고!!!
그리고 나온 값을 통해 수행하신 a/b test 결과를 설정하신 가설에 맞게 해석해주세요.

In [3]:
### A3-4. 코드작성

import pandas as pd
from scipy.stats import chi2_contingency

# 1. 관측 데이터 입력 (무한 스크롤 실험 결과라고 가정)
# Control(A): 기존 스크롤 방식 [클릭 11500, 미클릭 34500] -> 총 46,000명 (CTR 25.0%)
# Variant(B): 무한 스크롤 방식 [클릭 12190, 미클릭 33810] -> 총 46,000명 (CTR 26.5%)
control_results = [11500, 34500]
variant_results = [12190, 33810]

# 2. 분할표(Contingency Table) 생성
data = [control_results, variant_results]

# 3. 카이제곱 검정 수행
chi2, p_value, dof, expected = chi2_contingency(data)

# 4. 결과 출력
print(f"=== 무한 스크롤 A/B Test 결과 ===")
print(f"A그룹(기존) 클릭률: {control_results[0] / sum(control_results):.2%}")
print(f"B그룹(무한) 클릭률: {variant_results[0] / sum(variant_results):.2%}")
print(f"-----------------------")
print(f"카이제곱 통계량: {chi2:.4f}")
print(f"P-value: {p_value:.10f}")

alpha = 0.05
print(f"-----------------------")
if p_value < alpha:
    print(f"결과: P-value가 {alpha}보다 작으므로 귀무가설을 기각합니다.")
    print("해석: 무한 스크롤 도입이 공모전 클릭률에 통계적으로 유의미한 긍정적 변화를 주었습니다.")
else:
    print(f"결과: P-value가 {alpha}보다 크므로 귀무가설을 기각할 수 없습니다.")
    print("해석: 무한 스크롤 도입에 따른 클릭률 차이가 통계적으로 증명되지 않았습니다.")

#############

=== 무한 스크롤 A/B Test 결과 ===
A그룹(기존) 클릭률: 25.00%
B그룹(무한) 클릭률: 26.50%
-----------------------
카이제곱 통계량: 26.9884
P-value: 0.0000002047
-----------------------
결과: P-value가 0.05보다 작으므로 귀무가설을 기각합니다.
해석: 무한 스크롤 도입이 공모전 클릭률에 통계적으로 유의미한 긍정적 변화를 주었습니다.


**결과해석**

답변:통계적으로 유의미한 수준에서 귀무가설을 기각합니다. 즉, 무한 스크롤 형태로의 변경은 사용자의 공모전 탐색 및 클릭을 유도하는 데 효과가 있었다고 판단할 수 있습니다. 따라서 B안을 전체 배포하는 것을 제안합니다

<br>

**1주차 기초통계 과제는 여기까지였습니다. 수고하셨습니다!**